build_projects/dotnet-cli-build/GenerateBuildVersionInfo.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.DotNet.Cli.Build
 {
-    public class GenerateBuildVersionInfo : Task
+    public class GenerateBuildVersionInfo : ToolTask
```

Comment:
@eerhardt made a similar comment

ToolTask is so we can have msbuild invoke the git command for us. I combined this with the version generator to avoid passing commit count to it later.

My plan is to further decompose our version generation logic in a subsequent PR. This one just got big enough that I wanted to merge before further changing things. Once I'm done, we'll have tasks for both GIT assets, for doing MSI version bit-shifting, and I'll move branchinfo into a props file.